ios/CodePush/CodePush.m

```diff
@@ -458,11 +459,33 @@ - (void)savePendingUpdate:(NSString *)packageHash
                            rejecter:(RCTPromiseRejectBlock)reject)
 {
     NSError *error;
+    if (isRunningBinaryVersion) {
+        // isRunningBinaryVersion will not get set to "YES" if running against the packager.
+        NSString *binaryHash = [CodePushUpdateUtils getHashForBinaryContents:[CodePush binaryBundleURL] error:&error];
+        if (error) {
+            NSLog(@"Error obtaining hash for binary contents: %@", error);
+            resolve(nil);
```

Comment:
This behavior is kind of add. Now this method can return null, the metadata about a CodePush update, or the metadata about the binary contents. Should we change it to just always return an object that is either the binary or a CodePush update, and then reject the promise in error conditions?